### PR TITLE
Vendor updates

### PR DESCRIPTION
**Pull request by kinarashah**
Rebased after pulling in changes from https://github.com/rancher/machine/pull/17 + updated vendor to the latest version 

https://github.com/rancher/rancher/issues/23350

### CODE REVIEW

File: drivers/amazonec2/amazonec2_test.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package amazonec2
 
 import (
+	"github.com/Azure/go-autorest/autorest/to"
```
Comment (rmweir) [inline drivers/amazonec2/amazonec2_test.go]: should be with other github imports

File: drivers/azure/azure.go [hunk 2]
```diff
@@ -195,29 +192,6 @@ func (d *Driver) GetCreateFlags() []mcnflag.Flag {
...
-			Name:   flAzureFaultDomainCount,
-			Usage:  "Fault domain count to use for availability set",
-			EnvVar: "AZURE_FAULT_DOMAIN_COUNT",
-			Value:  3,
-		},
-		mcnflag.IntFlag{
-			Name:   flAzureUpdateDomainCount,
-			Usage:  "Update domain count to use for availability set",
-			EnvVar: "AZURE_UPDATE_DOMAIN_COUNT",
-			Value:  5,
```
Comment (rmweir) [inline drivers/azure/azure.go]: I'd imagine we still need these and want them to be configurable

File: drivers/azure/util.go [hunk 3]
```diff
@@ -2,12 +2,14 @@ package azure
 
 import (
 	"fmt"
```
Comment (rmweir) [inline drivers/azure/util.go]: imports should be separated

**Comment by daxmc99** [comment]
We upgraded this repository to use go modules, can you rebase your PR?

**Comment by deniseschannon** [comment]
Closing in favor of https://github.com/rancher/machine/pull/72

Available as of v0.15.0-rancher36

**Pull request closed without merging**